pyglet/util.py

```diff
@@ -16,25 +18,40 @@ def asbytes(s):
         return bytes(s)
 
 
-def asbytes_filename(s):
-    if isinstance(s, bytes):
+def asbytes_filename(s: Union[str, Buffer]) -> Optional[bytes]:
+    if isinstance(s, (bytes, bytearray, memoryview)):
         return s
     elif isinstance(s, str):
         return s.encode(encoding=sys.getfilesystemencoding())
 
+    return None
 
-def asstr(s):
+def asstr(s: Optional[Union[str, Buffer]]) -> str:
     if s is None:
         return ''
     if isinstance(s, str):
         return s
-    return s.decode("utf-8")
+    return s.decode("utf-8")  # type: ignore
+
+
+# Keep these outside of the function since we don't need to re-define
+# the function each time we make a call since no state is persisted.
+def _debug_print_real(arg: str) -> bool:
+    print(arg)
+    return True
+
+
+def _debug_print_dummy(arg: str) -> bool:
+    return True
+
 
+def debug_print(pyglet_option_name: str = 'debug') -> DebugPrintCallable:
```

Comment:
I would just put the actual return type, not an alias. `Callable[[str], bool]`

Reply:
I'll revert for now. If we need the type later as part of test / debug reworks, I'll add it back.